### PR TITLE
Password Reset

### LINKED ISSUE #76: Password reset process
**Is your feature request related to a problem? Please describe.**
The user lose the account forever if they forget their password.

**Describe the solution you’d like**
A password reset process, where the user can be sent an email that prompts of a new password.

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
image
image

**Features**
- Password Reset process
  Resolves #76
  - Forgot Password Dialog (initiates process)
  - Check Email Dialog (prompts the user to check email)
  - Forgot Password page (sets new password)
- Log In graphic

**Fixes**
- Fix for removing vertical scrollbar during animating in Schedule Periods disables full horizontal scrolling on mobile

**Notes to QA**
You need to temporarily change the Site URL configuration in our Supabase project. Go to Authentication > Settings > User Sessions > Site URL. Don’t forget to change it back when you’re done testing.

### CODE REVIEW

**Review by Temp9699 — APPROVED** [review]
lgtm

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -48,6 +52,14 @@ const App = ({
   // Query client
   const [queryClient] = useState(() => new QueryClient());
 
+  // Authentication
+  const router = useRouter();
+  useEffect(() => {
+    supabase.auth.onAuthStateChange((event) => {
+      if (event == "PASSWORD_RECOVERY") router.push("/account/forgot-password");
```
Comment (Jimmy-Tempest) [inline src/pages/_app.tsx:59]: What would happen if someone were to go to the route with an URL?

**Review by Jimmy-Tempest — APPROVED** [review]
LGTM!!

File: src/pages/_app.tsx (hunk 1 above)
Comment (SiravitPhokeed) [reply src/pages/_app.tsx:59]: At review time, this is what would happen:
- Not logged in, password recovery not declared: see page but can't interact
- Not logged in, password recovery declared: can use the page to set new password
- Logged in: can use the page to set new password (without going through the email process)

Logged in users should not be able to exploit the Password Reset process to change their password without any form of authentication, so I pushed a fix.

**Comment by SiravitPhokeed** [comment]
I just realized that I accidentally removed access to the Change Password Dialog; do we fix that now?

**Comment by Jimmy-Tempest** [comment]
> I just realized that I accidentally removed access to the Change Password Dialog; do we fix that now?

I think so

**Pull request merged**